client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/dynamosa/graphs/cdg/ControlDependenceGraph.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Adapted from the EvoSuite project (https://github.com/EvoSuite/evosuite)
+ * and modified for use in EvoMaster's Dynamosa module.
+ */
+package org.evomaster.client.java.instrumentation.dynamosa.graphs.cdg;
+
+import org.evomaster.client.java.instrumentation.dynamosa.graphs.EvoMasterGraph;
+import org.evomaster.client.java.instrumentation.dynamosa.graphs.cfg.*;
+import org.evomaster.client.java.utils.SimpleLogger;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public class ControlDependenceGraph extends EvoMasterGraph<BasicBlock, ControlFlowEdge> {
+
+    private final ActualControlFlowGraph cfg;
+
+    private final String className;
+    private final String methodName;
+
+    /**
+     * <p>Constructor for ControlDependenceGraph.</p>
+     *
+     * @param cfg a {@link org.evomaster.client.java.instrumentation.dynamosa.graphs.cfg.ActualControlFlowGraph} object.
+     */
+    public ControlDependenceGraph(ActualControlFlowGraph cfg) {
+        super(ControlFlowEdge.class);
+
+        this.cfg = cfg;
+        this.className = cfg.getClassName();
+        this.methodName = cfg.getMethodName();
+
+        computeGraph();
+    }
+
+    
+    /**
+     * Returns a Set containing all Branches the given BasicBlock is control
+     * dependent on.
+     * <p>
+     * This is for each incoming ControlFlowEdge of the given block within this
+     * CDG, the branch instruction of that edge will be added to the returned
+     * set.
+     *
+     * @param insBlock a {@link org.evomaster.client.java.instrumentation.dynamosa.graphs.cfg.BasicBlock} object.
+     * @return a {@link java.util.Set} object.
+     */
+    public Set<ControlDependency> getControlDependentBranches(BasicBlock insBlock) {
+        if (insBlock == null)
+            throw new IllegalArgumentException("null not accepted");
+        if (!containsVertex(insBlock))
+            throw new IllegalArgumentException("unknown block: " + insBlock.getName());
+
+        if (insBlock.hasControlDependenciesSet())
+            return insBlock.getControlDependencies();
+
+        Set<ControlDependency> r = retrieveControlDependencies(insBlock,
+                new LinkedHashSet<>());
+
+        return r;
+    }
+
+    private Set<ControlDependency> retrieveControlDependencies(BasicBlock insBlock,
+                                                               Set<ControlFlowEdge> handled) {
+
+        Set<ControlDependency> r = new LinkedHashSet<>();
+
+        for (ControlFlowEdge e : incomingEdgesOf(insBlock)) {
+            if (handled.contains(e))
+                continue;
+            handled.add(e);
+
+            ControlDependency cd = e.getControlDependency();
+            if (cd != null)
+                r.add(cd);
+            else {
+                BasicBlock in = getEdgeSource(e);
+                if (!in.equals(insBlock))
+                    r.addAll(retrieveControlDependencies(in, handled));
```

Comment:
put brackets for `if` even if single line. makes code easier to read, especially when nesting